### DISCUSSION TITLE
osm template files are not appearing on my Android device

### DISCUSSION THREAD

**Discussion by bellis89** [OpenOrienteering/mapper#2070] [Questions & Answers]
When I import an .osm file as a template for a new OOM map I can see it in OOM on my desktop, but when I transfer the same file(s) onto my Samsung TabA8 tablet, the osm image does not appear. The osm file is listed in the templates list on the tablet, but the font has a lower intensity. 
What can I do to make the osm template image visible in the Android system?

**Comment by pkturner** [comment]
It's not clear that you transferred both the .omap and the .osm files to your tablet.

When you open a .osm file as a template, Mapper does not _import_ the OSM data into the .omap. Mapper saves the location of the template (.osm) file so that it can find the data again, the next time the .omap is opened. It saves the relative path as one possible way to find the template file.

What I usually do with a template that I create on the desktop is to create a 'templates' folder within the folder that holds my .omap file. I keep the template (.osm) file in the 'templates' folder. Then on the tablet I also create a 'templates' folder in the same place that I copied the .omap file to. I copy the template (.osm) file from the desktop's 'templates' to the tablet's 'templates'.

Mapper tries several ways to find the template file on the tablet, but it is not foolproof.

**Comment by bellis89** [comment]
Thanks, Scott.
I do transfer both the omap and osm (and other template) files from my computer into the OOMapper folder on my tablet, so they are all in the same location in the memory. When I open the omap file on the tablet, all the other template files appear; only the osm file fails to appear, even though it shows up in the list of templates on the tablet.
It is not obvious to me that this is fundamentally different from the process you are describing; am I missing something?

I don’t normally do fieldwork using the osm template on my tablet so maybe I have not noticed this issue before, but I am currently mapping a complex heavily treed urban area where the street/sidewalk centre lines would be very helpful.
Brian

**Comment by pkturner** [comment]
Brian, in my previous answer, I mostly assumed that Mapper can use .osm template files just as effectively as it uses other templates. I have not used .osm templates in the field myself. All I have done with .osm data is import it into the .omap.

I would like to try to reproduce your issue. It should be simple enough, but I will not have time for it during the next day or two.

What is your desktop OS?

**Comment by bellis89** [comment]
I’m using Windows 11 on a Lenovo laptop, Scott.
Brian



On May 12, 2022, at 6:16 PM, Scott Turner ***@***.***> wrote:

﻿
[CAUTION: Non-UBC Email]

Brian, in my previous answer, I mostly assumed that Mapper can use .osm template files just as effectively as it uses other templates. I have not used .osm templates in the field myself. All I have done with .osm data is import it into the .omap.

I would like to try to reproduce your issue. It should be simple enough, but I will not have time for it during the next day or two.

What is your desktop OS?

**Comment by pkturner** [comment]
Thanks @bellis89. I reproduced your experience with a template added on a Windows desktop as well as on a Linux desktop. I don't have a guess as to what the limitation is which causes Mapper not to load the .osm template on Android.

As a workaround, you can create a .omap file containing just the OpenStreetMap data by opening the .osm file directly in Mapper. Then save it as a new .omap. You can then add the .omap as a template in the map that you're developing, and it will show up on Android.

**Comment by bellis89** [comment]
That seems like a smart workaround, Scott. I think I’ve done something similar when prepping maps for use with MapRun6.
Ultimately, I don’t really care exactly why the Android OOM can’t display an osm file, as long I can find a way to see it 😁
Thanks for your help!
Brian

On May 14, 2022, at 6:55 PM, Scott Turner ***@***.***> wrote:

﻿
[CAUTION: Non-UBC Email]

Thanks @bellis89<https://github.com/bellis89>. I reproduced your experience with a template added on a Windows desktop as well as on a Linux desktop. I don't have a guess as to what the limitation is which causes Mapper not to load the .osm template on Android.

As a workaround, you can create a .omap file containing just the OpenStreetMap data by opening the .osm file directly in Mapper. Then save it as a new .omap. You can then add the .omap as a template in the map that you're developing, and it will show up on Android.
